dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -101,39 +104,58 @@ def manual_pass
     redirect_to :manual_pass_form
   end
 
+  # get /admin/permissions
   def permissions_form
+    search_term = params[:search_term]
+    permission = params[:permission]
+    if search_term.present?
+      if search_term =~ /^\d+$/
+        @user = User.find(search_term)
+      else
+        users = User.where(hashed_email: User.hash_email(search_term))
+        @user = users.first
+        if users.count > 1
+          flash[:notice] = "More than one User matches email address.  "\
+                         "Showing first result.  Matching User IDs - #{users.map(&:id).join ','}"
+        end
+      end
+      unless @user || search_term.blank?
+        flash[:notice] = "User Not Found"
+      end
+    elsif permission.present?
+      @users_with_permission = User.joins(:permissions).where(user_permissions: {permission: permission}).order(:email)
+      @users_with_permission = @users_with_permission.page(page).per(page_size)
+    end
   end
 
-  # Grants the indicated permission to the indicated user. Expects params[:email] and
-  # params[:permission] to be populated.
   def grant_permission
-    permission = params[:permission]
-    user = User.find_by_email_or_hashed_email params[:email]
-
-    unless user && user.teacher?
-      flash[:alert] = "FAILED: user #{params[:email]} could not be found or was not a teacher"
-      redirect_to :permissions_form
+    user_id = params[:user_id]
+    @user = User.find(user_id)
```

Comment:
as Asher stated above, [find](https://apidock.com/rails/ActiveRecord/FinderMethods/find) will raise an `ActiveRecord::RecordNotFound` error, not return nil. I think you want `find_by(id: user_id)`.